cleanlab_pinecone_rag_workflow/Cleanlab_Pinecone_RAG.ipynb

```diff
@@ -0,0 +1,2673 @@
+   "id": "f07a408d",
+   "metadata": {},
+   "source": [
+    "We can see below that the number of chunks per document varies:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 139,
+   "id": "94377992",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "filename\n",
+      "documents/0-cognism-how-to-sell.pdf                 29.0\n",
+      "documents/01-danby-3cuft-range.pdf                   5.0\n",
+      "documents/02-Internet_and_Technology_Policy.docx     5.0\n",
+      "documents/03-job-offer-worksheet-aug-2019.docx       7.0\n",
+      "documents/04-FY24-Q1-NIKE-Press-Release.pdf         26.0\n",
+      "                                                    ... \n",
+      "documents/unit04.pdf                                22.0\n",
+      "documents/v0_FarmFreshFeast_financials.docx          7.0\n",
+      "documents/v0_FarmFreshFeast_financials.pdf           7.0\n",
+      "documents/what-is-selling.pptx                       7.0\n",
+      "documents/work-ez-specs.pdf                          2.0\n",
+      "Name: chunk_index, Length: 136, dtype: float64\n"
+     ]
+    }
+   ],
+   "source": [
+    "chunks_per_doc = chunk_df.groupby('filename')['chunk_index'].max() + 1\n",
+    "print(chunks_per_doc)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a867163e-9f41-46b8-9dca-055e5593f6bc",
+   "metadata": {},
+   "source": [
+    "## Use Cleanlab Studio to ensure there is no low quality or unacceptable text in our documents (e.g. PII, duplicates, non-English text, toxic language, etc.)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "98708651",
+   "metadata": {},
+   "source": [
+    "Cleanlab Studio runs many analyses of your dataset to find data issues and provide other useful metadata, including detection of: label issues, outliers, ambiguous examples, well-labeled examples, and more. For this particular use case, we will focus on text-specific data issues that Cleanlab can detect. After Cleanlab Studio runs these analyses on your data, new metadata columns are available to show the issue types detected and the severity of the issue type.\n",
+    "\n",
+    "We will first ingest the document chunks data obtained from our Pinecone DB into Cleanlab Studio and then run a project to analyze this data and look for the text issue types mentioned. After the project results are ready, we can make use of the `download_cleanlab_columns()` column to download the added metadata columns from the Cleanlab Studio project results.\n",
+    "\n",
+    "For more details on Cleanlab Studio Metadata you can go [here](https://help.cleanlab.ai/guide/concepts/cleanlab_columns/)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 143,
+   "id": "629b7d20-8e85-4476-94fe-c00c90a56f94",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from cleanlab_studio import Studio\n",
+    "\n",
+    "API_KEY = \"YOUR API KEY HERE\"\n",
```

Comment:
Nit: probably good to do another `if not os.getenv("CLEANLAB_API_KEY")`. Or at least name the API key e.g. `CLEANLAB_API_KEY = <YOUR API KEY HERE`

In general I feel that when our cleanlab studio notebooks are always having the API key in the notebook like this, it makes us seem like rubes, but I could be the exception